### PR TITLE
ANW-521

### PR DESCRIPTION
**Pull request by avatar382**
Javascript changes to support the enhancement request -- for subjects, a class is added to the element containing the icon to update the icon based on the subject term type.

## Description
Added CSS definitions for icons based on term type

In linker.js:
Javascript changes to add distinguishing class on subject search
Javascript changes to add distinguishing class on subject addition
Javascript changes to add distinguishing class for subjects on page load

## Related JIRA Ticket or GitHub Issue
https://archivesspace.atlassian.net/browse/ANW-521

## Motivation and Context
UI Enhacement, makes it possible to choose a subject with greater control without the browser

## How Has This Been Tested?
Manually tested the following use cases (latest Firefox on MacOS):

- The correct icons show when searching for subjects
- The correct icons show when adding subjects
- The correct icons show for subjects when page loads 

## Screenshots (if appropriate):
screen shot 2018-11-26 at 5 53 50 pm

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by avatar382** [comment]
@lorawoodford I replicated what you were seeing, with the updated icons not showing in the subject section in the show view for resources.

This is not a browser issue after all. I was confused because I thought you were referring to the edit page before the agents linker component is loaded. 

I'm working on a fix for the show view.

**Comment by avatar382** [comment]
@lorawoodford The correct icons for subjects are now showing in the show view for resources. This is ready for review again.

File: frontend/app/assets/stylesheets/archivesspace/token-input.less [hunk 1]
```diff
@@ -304,6 +304,52 @@ div.token-input-dropdown ul li.token-input-selected-dropdown-item {
     content:"\e041";
   }
 
+  // ANW-521
+  // Definitions for icons for subjects, based on the term type 
+  .subject .subject_type_cultural_context:before {
```
Comment (lorawoodford) [inline frontend/app/assets/stylesheets/archivesspace/token-input.less]: @avatar382 Forgive me if I'm dense, but looks like you've got duplicated `subject_type_cultural_context:before` here (with different content).

File: frontend/spec/selenium/spec/factories.rb [hunk 2]
```diff
@@ -0,0 +1,270 @@
...
+        file_format_name { generate(:file_format_name) }
+        file_format_version { generate(:alphanumstr) }
+        file_size_bytes { generate(:number).to_i }
+        checksum { generate(:alphanumstr) }
+        checksum_method { generate(:checksum_method) }
+      end
+
+
+      factory :extent, class: JSONModel(:extent) do
+        portion "whole"
```
Comment (lorawoodford) [inline frontend/spec/selenium/spec/factories.rb]:
```suggestion
        portion { "whole" }
```

File: frontend/spec/selenium/spec/factories.rb [hunk 3]
```diff
@@ -0,0 +1,270 @@
...
+        number "1"
```
Comment (lorawoodford) [inline frontend/spec/selenium/spec/factories.rb]:
```suggestion
        number { "1" }
```

File: frontend/spec/selenium/spec/factories.rb [hunk 4]
```diff
@@ -0,0 +1,270 @@
...
+        extent_type "gigabytes"
```
Comment (lorawoodford) [inline frontend/spec/selenium/spec/factories.rb]:
```suggestion
        extent_type { "gigabytes" }
```

File: frontend/spec/selenium/spec/factories.rb [hunk 5]
```diff
@@ -0,0 +1,270 @@
...
+
+      factory :name_person, class: JSONModel(:name_person) do
+        rules { generate(:name_rule) }
+        source { generate(:name_source) }
+        primary_name { generate(:generic_name) }
+        rest_of_name { generate(:generic_name) }
+        sort_name { generate(:sort_name) }
+        name_order { %w(direct inverted).sample }
+        number { generate(:alphanumstr) }
+        sort_name_auto_generate true
```
Comment (lorawoodford) [inline frontend/spec/selenium/spec/factories.rb]:
```suggestion
        sort_name_auto_generate { true }
```

**Comment by avatar382** [comment]
@lmcglohon @lorawoodford Good catch on the documentation, and the duplication between the old selenium tests and these. 

My understanding is that the selenuim:staff tests are so brittle that they are not run when vetting a release, and need to to be refactored and/or removed. 

The intention behind these frontend:selenium tests was to have a way of "unit testing" the frontend UI. I've put the words unit testing in quotes because they are integration tests between the frontend JS and Rails views/controllers. Still, the idea is that these frontend:selenium tests cover single actions like a unit test would.

I think quite a bit of the old selenium:staff tests fit that scope and can be refactored or moved into the frontend:selenium suite. 

So I'm thinking one option is to move the selenium:staff tests out of master and into another branch. Since they're not being used, that will help clean stuff up in the short term. Then, whatever tests in selenium:staff that we want to keep can be refactored into frontend:selenium. ANW-815 covers setting up frontend:selenium, so it seems like a good place for this work.

I'm happy to update the docs once our direction on this becomes clearer.

File: frontend/app/assets/stylesheets/archivesspace/token-input.less (hunk 1 above)
Comment (avatar382) [reply frontend/app/assets/stylesheets/archivesspace/token-input.less]: Yep, that's a typo! I'll push a fix shortly.

**Comment by lmcglohon** [comment]
@lorawoodford @avatar382 I have updated the tech-docs repo by removing the selenium README and other references to it. Once we have merged this PR into master, I will make sure to update the tech-docs to reflect these new tests. I think @lorawoodford has a great suggestion that we move the selenium drivers into the common directory so they can be used by both the frontend and public selenium tests. @avatar382 what would it take to make that move for the new tests you are adding here?

**Comment by avatar382** [comment]
@lorawoodford @lmcglohon I've moved the selenium geckodriver binary into it's own place in the common directory. Will Travis automatically run the new frontend selenium tests, or does it have to be configured to do so?

**Comment by lmcglohon** [comment]
@avatar382 the frontend:selenium tests will not run automatically by travis. We need to configure the travis tests to include them.

**Comment by lorawoodford** [comment]
Thanks for this, @avatar382 .  Looks good to go from me, so merging.  @lmcglohon good to go to update tech-docs and add to the travis tests.

Also, I'm gonna go PR an update to the public tests to point to this set of drivers and dump the duplicate drivers in /selenium.
